src/distributed_physical_optimizer_rule.rs

```diff
@@ -376,14 +395,17 @@ enum DistributedPlanError {
     /// Prompts the planner to limit the amount of tasks used in the stage that is currently
     /// being planned.
     LimitTasks(usize),
+    /// Signals the planner that this whole plan is non-distributable. This can happen if
+    /// certain nodes are present, like [StreamingTableExec], which are typically used in
+    /// queries that rather performing some execution, they perform some introspection.
+    NonDistributable,
```

Comment:
we can put the plan name `StreamingTableExec::name` in here if it makes sense

Reply:
also it may be useful to expose an option for users to specify nodes which they do not want to be distributed